### PR TITLE
[WIP] Solr namespace for this hydra app (avoid collision with other hydra apps)

### PR DESCRIPTION
**Pull request by darrenleeweber**
hydra -> hyku

@projecthydra-labs/hyrax-code-reviewers

- might be rebased, pending #1052, which extracts commits here without the namespace change.

### CODE REVIEW

**Comment by darrenleeweber** [comment]
If coveralls coverage decreased in this PR, there must be some hard-coded strings in the specs somewhere, or some general assumptions about the solr namespace that need to change.  The namespace should be configurable.  Does this PR miss some config settings?

**Comment by mjgiarlo** [comment]
@darrenleeweber I believe some of our underlying components generate config files with `hydra-development` and `hydra-test` in them. I'm not sure why changing these values wouldn't work, though.

Thoughts, @jcoyne?

File: config/settings.yml [hunk 1]
```diff
@@ -16,8 +16,14 @@ action_controller:
...
 
+<%
+  # Read the config/solr.yml
+  # - note that it specifies a URL that includes the collection path, which is stripped below.
+  solr_config = YAML.load(ERB.new(IO.read(Rails.root + 'config' + 'solr.yml')).result)[Rails.env].with_indifferent_access
+%>
+
 solr:
-  url: "http://127.0.0.1:8983/solr/"
+  url:  <%= solr_config["url"].gsub(/solr\/.*/, 'solr/') %>
```
Comment (jcoyne) [inline config/settings.yml]: Settings.yml should be the authoritative source, don't change these

Comment (darrenleeweber) [inline config/settings.yml]: This concern to be addressed in #1052 and see what happens, this PR is pending that debate and needs to be rebased or whatever.

Comment (atz) [inline config/settings.yml]: I agree w/ @jcoyne here.  Also, the purpose of *this* PR is to do the namespacing.  If we can keep it separate from config consolidation, all the better.  If we can't (for testing consistency reasons), then we should prefer for Settings to be the authoritative source.

**Pull request closed without merging**